02-OOP-Inheritance/Ex05/src/classEventEmitter.js

```diff
@@ -0,0 +1,32 @@
+class eventEmitter {
+	constructor(events) {
+		this.events (eventName, callbacks) = {
+			this.eventName: this.eventName,
+			this.callbacks: [],
+		};
+	}
+	on(eventName, callback) {
```

Comment:
Then, when you wanna listen a particular event, you will add the new event as a key of `this.events` with the callback. If is the second time you are adding a listener for a particular time, you just push a new callback in the array of callbacks for this event.

```
on (eventName, callback) {
    let event = this.events[eventName];

    if (event) { // If that returns true, means that you already defined the array with the callbacks. At least, is the second time you are adding a callback for that event. 
        event.push(callback); //Just push one new callback into the array
    } else {
        event = [callback]; // If not, you will initialize the array with the first callback.
    }
}
```